python/tskit/metadata.py

```diff
@@ -120,6 +123,191 @@ def decode(self, data: bytes) -> bytes:
         return data
 
 
+def binary_format_validator(validator, types, instance, schema):
```

Comment:
This feels slightly strange to be yielding an exception instance. I guess this is the interface defined by upstream? Be good to note this.

Reply:
Comment added in https://github.com/tskit-dev/tskit/pull/601/commits/0fb1e66b68b8a060eb49ffaff527170cbc6a66f1